code-challenges.test.js

```diff
@@ -26,26 +30,194 @@ const colors2 = [
+      [newArray[i], newArray[shuffle]] = [newArray[shuffle], newArray[i]] 
+    } 
+    return newArray
+}
+
+// PASSED!
+// PASS  ./code-challenges.test.js
+// arrayContaining
+//   ✓ takes in an array and removes the first item from the array and shuffles the remaining content (1 ms)
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
+// Snapshots:   0 total
+// Time:        0.145 s, estimated 1 s
+
+
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const votes1 = { upVotes: 13, downVotes: 2 }
-// Expected output: 11
-const votes2 = { upVotes: 2, downVotes: 33 }
-// Expected output: -31
+// Process: RED!
+
+describe("votes", () => {
+  it("takes in an object that contains up votes and down votes and returns the end tally.", () => {
+    const votes1 = { upVotes: 13, downVotes: 2 }
+    // Expected output: 11
+    const votes2 = { upVotes: 2, downVotes: 33 }
+    // Expected output: -31
+    
+    // expect statement will contain the function call to call the given functions and;
+    // place test variable outputs inside the .toEqual() matcher parameter to compare the given outputs
+      expect(votes(votes1)).toEqual(11)
+      expect(votes(votes2)).toEqual(-31)
+  })
+})
+
+// FAIL!: output - votes is not defined
+// FAIL  ./code-challenges.test.js
+// votes
+//   ✕ takes in an object that contains up votes and down votes and returns the end tally. (1 ms)
+
+// ● votes › takes in an object that contains up votes and down votes and returns the end tally.
+
+//   ReferenceError: votes is not defined
+
 
 // b) Create the function that makes the test pass.
 
+// Process: GREEN!
+
+// PSEUDOCODE:
+// Create a function that takes in an object that contains up votes and down votes and returns the end tally.
+
+// Input: an object ("takes in..")
+// Output: contains up votes and down votes and returns the end tally.
+
+// Given objects:
+// Input: { upVotes: 13, downVotes: 2 }
+// Output: 11
+
+// Input: { upVotes: 2, downVotes: 33 }
+// Output: -31
+
+// Create a function named "votes" that takes in an object named "obj" that contains up votes and down votes and returns the end tally.
+//Create a function named "votesTally" that assigns the object method and key value of upVotes and use math logic to subtract the object and key value of downVotes that will equal the sum of the votesTally.
+//Return the end tally of the sum of all votes. 
+
+const votes = (obj) => { 
+  const votesTally = obj.upVotes - obj.downVotes
+  return votesTally
+}
+
+//PASSED!
+//  PASS  ./code-challenges.test.js
+// votes
+//   ✓ takes in an object that contains up votes and down votes and returns the end tally.
+
+// Test Suites: 1 passed, 1 total
+// Tests:       2 passed, 2 total
+// Snapshots:   0 total
+// Time:        0.129 s
+
+
```

Comment:
Great job! General rule of thumb, if you are creating a variable just to return it right after, you can skip the creation of the variable and just return the data itself!